index.js

```diff
@@ -192,78 +202,97 @@
   //  Value :: Location
   var Value = 'Value';
 
-  //  _funcPath :: (Boolean, Array String, a) -> Nullable Function
-  function _funcPath(allowInheritedProps, path, _x) {
-    var x = _x;
-    for (var idx = 0; idx < path.length; idx += 1) {
-      var k = path[idx];
-      if (x == null || !(allowInheritedProps || has(k, x))) return null;
-      x = x[k];
-    }
-    return typeof x === 'function' ? x : null;
+  //  prefix :: String
+  var prefix = 'fantasy-land/';
+
+  //  funcProp :: (String, a) -> Nullable Function
+  function funcProp(prop, x) {
+    return x != null && typeof x[prop] === 'function' ? x[prop] : null;
   }
 
-  //  funcPath :: (Array String, a) -> Nullable Function
-  function funcPath(path, x) {
-    return _funcPath(true, path, x);
+  //  functionImpl :: (String, String) -> Nullable Function
+  function functionImpl(type, name) {
+    return (implementations[type] != null &&
+            implementations[type][name]) || null;
```

Comment:
I find code clearer when an operator's operands are all on one line or begin at the same column on different lines:

```javascript
foo || bar
```

```javascript
some relatively long expression ||
another relatively long expression
```

_(I apply the same rule to function arguments.)_

My objection to the code above is that `|| null` can be misread as relating to `implementations[type][name]` rather than to the whole parenthesized expression.

This looks ugly, but I prefer it:

```javascript
return (implementations[type] != null &&
        implementations[type][name]) ||
       null;
```

Left to my own devices I might remove the type coercion:

```javascript
return implementations[type] != null &&
       implementations[type][name] != null ?
         implementations[type][name] :
         null;
```

I'm sure the inefficiency of this version would bother you, Aldwin. I'm happy to stick with the coercive version, but please update the indentation. :)